src/cast_core/src/CommitLatency.cpp

```diff
@@ -0,0 +1,227 @@
+                    }
+
+                    // result1 = db.hltest.find_one( { '_id': 1 }, session=session )
+                    bsoncxx::document::value doc_filter1 = bsoncxx::builder::stream::document{}
+                        << "_id" << 1
+                        << bsoncxx::builder::stream::finalize;
+                    auto result1 = config->collection.find_one(*_session, doc_filter1.view(), config->optionsFind);
+
+                    // db.hltest.update_one( {'_id': 1}, {'$inc': {'n': -amount}}, session=session )
+                    bsoncxx::document::value doc_update1 = bsoncxx::builder::stream::document{}
+                        << "$inc" << bsoncxx::builder::stream::open_document
+                            << "n" << -amount
+                        << bsoncxx::builder::stream::close_document
+                        << bsoncxx::builder::stream::finalize;
+                    config->collection.update_one(*_session, doc_filter1.view(), doc_update1.view(),
+                                                  config->optionsUpdate);
+
+                    // result2 = db.hltest.find_one( { '_id': 2 }, session=session )
+                    bsoncxx::document::value doc_filter2 = bsoncxx::builder::stream::document{}
+                        << "_id" << 2
+                        << bsoncxx::builder::stream::finalize;
+                    auto result2 = config->collection.find_one(*_session, doc_filter2.view(), config->optionsFind);
+
+                    // db.hltest.update_one( {'_id': 2}, {'$inc': {'n': amount}}, session=session )
+                    bsoncxx::document::value doc_update2 = bsoncxx::builder::stream::document{}
+                        << "$inc" << bsoncxx::builder::stream::open_document
+                            << "n" << amount
+                        << bsoncxx::builder::stream::close_document
+                        << bsoncxx::builder::stream::finalize;
+                    config->collection.update_one(*_session, doc_filter2.view(), doc_update2.view(),
+                                                  config->optionsUpdate);
+
+                    // result = db.hltest.aggregate( [ { '$group': { '_id': 'foo', 'total' : { '$sum': '$n' } } } ], session=session ).next()
+                    mongocxx::pipeline p{};
+                    bsoncxx::document::value doc_group = bsoncxx::builder::stream::document{}
+                            << "_id" << "foo"
+                            << "total" << bsoncxx::builder::stream::open_document
+                                << "$sum" << "$n"
+                            << bsoncxx::builder::stream::close_document
+                        << bsoncxx::builder::stream::finalize;
+                    p.group(doc_group.view());
+                    auto cursor = config->collection.aggregate(*_session, p, config->optionsAggregate);
+                    // Check for isolation or consistency errors
```

Comment:
[Question] If the following loop is used to check "isolation or consistency errors", did you want to wrap it with a `if (config->useSession)`? If not, suggest updating the comment.

Reply:
Thanks for the clarification. Maybe rename this comment to "Record isolation or consistency violations in non-transactional operations"? "Errors" sounds really ominous to me XD.

Reply:
I added a sentence explaining when they are expected.